src/core1/DynamicGenericArray.java

```diff
@@ -0,0 +1,51 @@
+package core1;
+
+public class DynamicGenericArray <T> {
+    private Object[] array;
+    private int currLength;
+
+    public DynamicGenericArray(int length) {
+        array = new Object[length];
+        currLength = 0;
+    }
+
+    public DynamicGenericArray() {
+        this(10);
+    }
+
+    public T get(int i) {
+        return (T)array[i];
+    }
+
+    public void add(T e) {
```

Comment:
Try to avoid "one-letter-variable". Nobody will scold you if you name variables with full names, especially if there are no JavaDocs :)
That's also about get() method.